content/dev_stories/playground_to_production_fruit_falling.md

```diff
@@ -0,0 +1,677 @@
+
+And with that, my entire prototype, both code and assets, was running 
+locally from my own repos where I could iterate fast and commit often
+as I moved on to...
+
+### Developing the Experience
+
+Coding tutorials and implementation diatribes are the stuff of dreams in 
+that most people who read them fall asleep almost immediately. This is not 
+a coding tutorial, so I won't spend much time describing the details of the
+implementation of *Fruit Fallin'*. I will, however, cover a few notes that
+I think were particularly important to helping me get *Fruit Fallin'* to
+work the way I wanted it.
+
+
+1.	I pulled apart and deleted `playground.ts` just as fast as I could.
+    
+    The Babylon.js Playground is a fantastic place to experiment, prototype,
+    and repro bugs. In order to be as good at those things as it is, 
+    however, the Playgrund make a number of implementation decisions that
+    do not translate well into production software development. Single-file
+    development, kitchen-sink dependency inclusion, and the `BABYLON` 
+    import are prominent examples of this.
+
+    ![The BABYLON import](/img/devVignettes/fruitFalling/16_babylon_import.jpg!350)
+
+    In its initial state, the NPM Package Template repo emulates these 
+    patterns intentionally in `playground.ts` to make it as easy as 
+    possible to transition from Playground-based prototyping to repo-based 
+    development. The first step of that transition is to simply copy the
+    Playground code into `playground.ts` as discussed above, then get that
+    to a working state. Once that's done, however, I think the second step
+    should always be to refactor everything in `playground.ts` into a more
+    robust code structure (multiple files, types, etc.). With the code 
+    already in a working state, I moved functionality out into separate 
+    files incrementally (camera creation into a new camera type, falling
+    objects into a new class, etc.), proving at every step that I hadn't
+    broken my code. Then, once all the functionality had been moved out
+    of `playground.ts`, I simply deleted that file and the folder 
+    containing it from my repo.
+1.  I cleaned out the unused dependencies from the NPM Package Template
+    repository.
+
+    This is another step I recommend for moving your new repo away from 
+    Playground practices and into production. In order to be able to 
+    support code copy-pasted out of the Playground, the NPM Package
+    Template's `app_package` by default includes a *lot* of dependencies,
+    only a subset of which are likely to be used by any particular 
+    codebase. Fortunately, it's easy to get rid of the unnecessary imports
+    here; all I had to do was open `app_package/package.json` and delete
+    the references to any NPM packages I wasn't using--which, in this case,
+    was most of them.

```

Comment:
Might be worth linking to the tree shaking doc? https://doc.babylonjs.com/divingDeeper/developWithBjs/treeShaking

Reply:
Is tree shaking involved in this? My recollection is that my build size didn't actually change when I did this, which I thought made sense because the tree shaking was causing things I wasn't using to get ignored anyway. Thus, removing these dependencies didn't have any effect on tree shaking, it just reduced the unnecessary overhead of the dev repo itself. @RaananW, have I understood that correctly? If so, I feel like bringing up tree shaking here might be a complicating detour.

Reply:
@syntheticmagus - I think you are right, it would over-complicate things to mention it. 
If you didn't use those imports before and just imported them, they would have been included in the package, but I can only assume the framework itself needs them (or you need them somewhere else in the code) and they are imported anyhow. 

> tree shaking was causing things I wasn't using to get ignored anyway

in an ideal world this is true, but in babylon, since side effects are turned on, importing already means "i need it" because there might be side effects in this class. So no packer will remove this from your packaged build.

Reply:
Ooooh, ok, I thought that could explain why taking the unnecessary imports out is good, but it would be a detour.